service-map/src/main/java/fi/nls/oskari/map/layer/formatters/LayerJSONFormatter.java

```diff
@@ -220,29 +219,36 @@ public static JSONObject createStylesJSON(String name, String title, String lege
     }
 
     // This is solution of transition for dataUrl and for dataUrl_uuid
-    private String getFixedDataUrl(final OskariLayer layer) {
-        final String metadataId = layer.getMetadataId();
+    protected static String getFixedDataUrl(String metadataId) {
         if(metadataId == null || metadataId.isEmpty()) {
             return null;
         }
-        if(metadataId.toLowerCase().startsWith("http")) {
-            try {
-                URL url = new URL(metadataId);
-
-                String[] parameters = url.getQuery().split("&");
-                for (String param : parameters) {
-                    String[] keyvalue = param.split("=");
-                    if("uuid".equalsIgnoreCase(keyvalue[0]) || KEY_ID.equalsIgnoreCase(keyvalue[0])) {
-                        return keyvalue[1];
-                    }
-                }
-            } catch (Exception ignored) {
-                // propably just not valid URL
+        if(!metadataId.toLowerCase().startsWith("http")) {
+            // not a url -> return as is
+            return metadataId;
+        }
+        try {
+            Map<String, List<String>> params = IOHelper.parseQuerystring(metadataId);
+            String idParam = params.keySet().stream()
```

Comment:
For fun `return IOHelper.parseQuerystring(metadataId).entrySet().stream().filter(e -> "uuid".equalsIgnoreCase(e.getKey()) || KEY_ID.equalsIgnoreCase(e.getKey())).map(e -> e.getValue()).findAny().map(it -> it.stream()).orElse(null);`